agent/agent_worker.go

```diff
@@ -362,18 +402,44 @@ func (a *AgentWorker) Heartbeat(ctx context.Context) error {
 // Performs a ping that checks Buildkite for a job or action to take
 // Returns a job, or nil if none is found
 func (a *AgentWorker) Ping(ctx context.Context) (*api.Job, error) {
-	ping, _, err := a.apiClient.Ping(ctx)
-	if err != nil {
+	ping, resp, pingErr := a.apiClient.Ping(ctx)
+	// wait a minute, where's my if err != nil block? TL;DR look for pingErr ~20 lines down
```

Comment:
Good explanation 👍

Is err renamed to pingErr to make finding it across 20 lines easier? (That's reasonable, mostly curious)

The other thing I was wondering, and this is not necessarily a suggestion here, but could the ergonomics of apiClient.Ping be changed to make the gap unnecessary? I can see lifting errUnrecoverable into the API client, but handling ping.Message and ping.Action is definitely too business-logicy for that layer, which is the issue.

Reply:
> Is err renamed to pingErr to make finding it across 20 lines easier?

got it in one 👍

> but could the ergonomics of apiClient.Ping be changed to make the gap unnecessary?

yeah, i investigated this a little bit, and couldn't find a way to make it nice. i think part of the issue is that the apiclient exposes non-2xx status codes as go errors, which is something that i think is a little weird, and makes the ergonomics of failure cases a little strange. changing that now would be a major PITA though, so i think it's probably best to leave it for now